File: Solutions/Java/Triplet_Sum_in_Array.java
class Solution
{
    //Function to find if there exists a triplet in the 
    //array A[] which sums up to X.
    public static boolean find3Numbers(int A[], int n, int X) { 
    
        // Your code Here
        int k = n-1, sum = 0, j = 0;
        Arrays.sort(A);
        for (int i=0; i<n-2; i++) {
            k = n-1; j = i+1;
            while (j<k) {
                sum = A[i] + A[j] + A[k];
                if (sum==X) {
                    return true;
                }
                else if (sum>X) {
                    k--;
                }
                else {
                    j++;
                }
            }
        }
        return false;
    }
}

File: Solutions/Python/Triplet_Sum_in_Array.py
class Solution:
     
    #Function to find if there exists a triplet in the 
    #array A[] which sums up to X.
    def find3Numbers(self,A, n, X):
        # Your Code Here
        k = n-1
        summ = 0 
        j = 0
        A.sort()
        for i in range(n-2):
            k = n-1
            j = i+1
            while j<k:
                summ = A[i] + A[j] + A[k]
                if summ == X:
                    return True
                elif summ > X:
                    k -= 1
                else:
                    j += 1
        return False
